lib/pengine/unpack.c

```diff
@@ -573,17 +574,28 @@ unpack_nodes(xmlNode * xml_nodes, pe_working_set_t * data_set)
             new_node = NULL;
 
             id = crm_element_value(xml_obj, XML_ATTR_ID);
+            cluster_id = crm_element_value(xml_obj, PCMK_XA_CLUSTER_ID);
             uname = crm_element_value(xml_obj, XML_ATTR_UNAME);
             type = crm_element_value(xml_obj, XML_ATTR_TYPE);
             score = crm_element_value(xml_obj, XML_RULE_ATTR_SCORE);
-            crm_trace("Processing node %s/%s", uname, id);
+
+            crm_trace("Unpacking node %s with cluster ID %s and name %s",
+                      pcmk__s(id, "<none>"), pcmk__s(cluster_id, "same as id"),
+                      pcmk__s(uname, "same as id"));
 
             if (id == NULL) {
                 pcmk__config_err("Ignoring <" XML_CIB_TAG_NODE
                                  "> entry in configuration without id");
                 continue;
             }
-            new_node = pe_create_node(id, uname, type, score, data_set);
+
+            if (cluster_id == NULL) {
+                cluster_id = id;
+            }
+            if (uname == NULL) {
+                uname = id;
+            }
+            new_node = pe_create_node(cluster_id, uname, type, score, data_set);
```

Comment:
I suppose the new `cluster-id` attribute can supplant all internal uses of `id` for nodes? I hope so... there's nowhere to put a new attribute. `pe_node_s` and `pe_node_shared_s` are both public API and aren't created by constructors.

To confirm: `id` is intended to be used only (1) for IDREFs and (2) as the default value for cluster-id and uname, correct? We should never need to use a node's `id` directly.